# PDVarFilter - Version 0.1
## Phenotype and Diagnosis level VARiant FILTERing tool for WGS/WES sequencing data.
### Introduction
#### PDVarFilter is a Phenotype and Diagnosis level VARiant FILTERing tool, which filter SNP/INDEL variants based on its presense in individulas of pedigree. This tool filter SNP/INDEL variants by referring Annovar tool text file, there are four three core criteria used for filtering: 1) Damaging variants based on in silico predictions algorithms (SIFT or PolyPhene) for SNP and Frameshift/Nonfremeshift/Stopgain/Stoploss varaitns for INDEL. 1) Rare variants (<=1%) in non finnish population (NFE) as per GnomAD database, and 3) Variants absent in control individulas of pedigree.

This tool has two main modules and variants filtered further based on Phenotype and Diagnosis level:
1) PhenoVarFilter
2) DiagnoVarFilter

PDVarFilter is developed as part of the Multiplex Autism Families study at ARC, University of Cambridge, UK. Please read the README file before using this tool.

### Requirements
PDVarFilter tool is tested on following platform:
1. OS Ubuntu v16.04.5 LTS
2. Perl 5, version 22, subversion 1 (v5.22.1)

### Usage
1) PhenoVarFilter:
```
Input files for snp and indel: (Please note that input files and edit in samples details is mendetory before running the tool)

1. Paste all samples annovar annotated text file in the folder 'input_annovar_annotated_txt_files' for snp and indel.

2. Edit files in the folder 'input_samples_details' for snp and indel.

File 'input_all_cases.txt' - Add tab delimited information of pedigree name, sample name, and full path of annotated annovar text files of cases in pedigree.

File 'input_bap.txt' - Add BAP samples name in list.

File 'input_map.txt' - Add MAP samples name in list.

File 'input_nap.txt' - Add NAP samples name in list.

File 'input_control.txt' - Add control samples name in list.

Output file: 

GF18.snp/indel.PhenoVarFilter.results - This output file contains the filtered snp/indel variants in BAP, MAP and NAP cateogory in pedigree.

Command syntax for SNP:
usage: PhenoVarFilter_snp.sh WORKING_DIRECTORY_PATH PEDIGREE_NAME TOTAL_BAP_SAMPLES TOTAL_MAP_SAMPLES TOTAL_NAP_SAMPLES

arguments:
  WORKING_DIRECTORY_PATH        Path to the working directory where files are stored.
  PEDIGREE_NAME        Pedigree name of samples.
  TOTAL_BAP_SAMPLES        Total number of BAP individuals in pedigree
  TOTAL_MAP_SAMPLES        Total number of MAP individuals in pedigree
  TOTAL_NAP_SAMPLES        Total number of NAP individuals in pedigree

Example command: 
cd /usr/ravi/PDVarFilter/PhenoVarFilter/snp
/usr/ravi/PDVarFilter/PhenoVarFilter/snp/PhenoVarFilter_snp.sh /usr/ravi/PDVarFilter/PhenoVarFilter/snp GF18 3 2 2

For INDEL:

usage: PhenoVarFilter_indel.sh WORKING_DIRECTORY_PATH PEDIGREE_NAME TOTAL_BAP_SAMPLES TOTAL_MAP_SAMPLES TOTAL_NAP_SAMPLES

arguments: same as snp

Example command: 
cd /usr/ravi/PDVarFilter/PhenoVarFilter/indel
/usr/ravi/PDVarFilter/PhenoVarFilter/indel/PhenoVarFilter_indel.sh /usr/ravi/PDVarFilter/PhenoVarFilter/indel GF18 3 2 2

```

1) DiagnoVarFilter:

```

Instructions
The folder "example_files" contains a set of input files in the required format for varPrio. The variants file contain variants only from chromosome 19 as an example. This folder also contains output files generated by varPrio.

Create a file detailing the information about input variant files (INPUTFILEINFO) This file contains 3 rows. 1st row - Sample identifier of the affected individuals; 2nd row - Family identifier; 3rd row - Path to the annotated file (ANNOVAR tab delimmited TXT files). This program is tailor-made for large-scale analysis of pedigrees recruited in ADBS. The input formats recognized by this tool is based on the files generated in ADBS. This tool is not generalized to read any type of annotated VCFs.

Provide counts of variants in population controls and familial controls (POPULATIONCONTROL and ALLFAMILYCONTROL) These files 3 rows: chr, pos and count

Create output directory in which you need varPrio to write the results to.

Example commands
mkdir ./example_files/output_snp
mkdir ./example_files/output_indel

python varprio-0.4.py -T snp \
	-I /home/husayn/varPrio-0.4/example_files/input_info_snp.txt \
	-PC /home/husayn/varPrio-0.4/example_files/INDEX-db_phase1_snp_population_control_chr19.txt \
	-AFC /home/husayn/varPrio-0.4/example_files/All_fam_control_count.txt \
	-O /home/husayn/varPrio-0.4/example_files/output_snp 

python varprio-0.4.py -T indel \
	-I /home/husayn/varPrio-0.4/example_files/input_info_indel.txt \
	-PC /home/husayn/varPrio-0.4/example_files/INDEX-db_phase1_indel_population_control_chr19.txt \
	-AFC /home/husayn/varPrio-0.4/example_files/All_fam_control_count.txt \
	-O /home/husayn/varPrio-0.4/example_files/output_indel 
Output files
Results of every step is written to a separate file. This helps in customizing prioritization approach as per the requirement.

In the case of SNP, the final files are "LIST2A_step3_1to5P_withPCAFC.vpr" and "LIST2B_step3_1to5P_withPCAFC.vpr". These contain prioritized variants as described above.

Five new columns are added to the output files. These contain sampleID, pedigreeID, number of algorithms calling it damaging, occurrence/count in population controls and occurrence/count in familial controls respectively.

While the LIST2B contains all columns provided by the ANNOVAR annotation, LIST2A contains only selected columns required in the context of ADBS downstream analysis.

In the case of INDELs, "step2_prioritized_INDEL_LIST3.vpr" is the final prioritized list of variants. Three new columns are added in the output files, containing sampleID, pedigreeID, presence/absence in the population controls.

Column headers of "LIST2B_step3_1to5P_withPCAFC.vpr":
Chr Start End Ref Alt Func.refGene Gene.refGene GeneDetail.refGene ExonicFunc.refGene AAChange.refGene cytoBand genomicSuperDups esp6500siv2_all 1000g2015aug_all 1000g2015aug_eur ExAC_ALL ExAC_AFR ExAC_AMR ExAC_EAS ExAC_FIN ExAC_NFE ExAC_OTH ExAC_SAS avsnp147 SIFT_score SIFT_pred Polyphen2_HDIV_score Polyphen2_HDIV_pred Polyphen2_HVAR_score Polyphen2_HVAR_pred LRT_score LRT_pred MutationTaster_score MutationTaster_pred MutationAssessor_score MutationAssessor_pred FATHMM_score FATHMM_pred PROVEAN_score PROVEAN_pred VEST3_score CADD_raw CADD_phred DANN_score fathmm-MKL_coding_score fathmm-MKL_coding_pred MetaSVM_score MetaSVM_pred MetaLR_score MetaLR_pred integrated_fitCons_score integrated_confidence_value GERP++_RS phyloP7way_vertebrate phyloP20way_mammalian phastCons7way_vertebrate phastCons20way_mammalian SiPhy_29way_logOdds Otherinfo1 Otherinfo2 Otherinfo3 Otherinfo4 Otherinfo5 Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13 Sample_ID Pedigree_ID Predicted_deleterious_by PC_Count AFC_Count

Column headers of "LIST2A_step3_1to5P_withPCAFC.vpr":
Chr Start Ref Alt Func.refGene Gene.refGene ExonicFunc.refGene AAChange.refGene 1000g2015aug_all ExAC_ALL ExAC_SAS avsnp147 SIFT_pred Polyphen2_HDIV_pred LRT_pred MutationTaster_pred MutationAssessor_pred Sample_ID Pedigree_ID Predicted_deleterious_by PC_Count AFC_Count

Column headers of "step2_prioritized_INDEL_LIST3.vpr":
Chr Start End Ref Alt Func.refGene Gene.refGene GeneDetail.refGene ExonicFunc.refGene AAChange.refGene cytoBand genomicSuperDups esp6500siv2_all 1000g2015aug_all 1000g2015aug_eur ExAC_ALL ExAC_AFR ExAC_AMR ExAC_EAS ExAC_FIN ExAC_NFE ExAC_OTH ExAC_SAS avsnp147 SIFT_score SIFT_pred Polyphen2_HDIV_score Polyphen2_HDIV_pred Polyphen2_HVAR_score Polyphen2_HVAR_pred LRT_score LRT_pred MutationTaster_score MutationTaster_pred MutationAssessor_score MutationAssessor_pred FATHMM_score FATHMM_pred PROVEAN_score PROVEAN_pred VEST3_score CADD_raw CADD_phred DANN_score fathmm-MKL_coding_score fathmm-MKL_coding_pred MetaSVM_score MetaSVM_pred MetaLR_score MetaLR_pred integrated_fitCons_score integrated_confidence_value GERP++_RS phyloP7way_vertebrate phyloP20way_mammalian phastCons7way_vertebrate phastCons20way_mammalian SiPhy_29way_logOdds Otherinfo1 Otherinfo2 Otherinfo3 Otherinfo4 Otherinfo5 Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13 Sample_ID Family PC

Contact
For technical queries, please write to Dr. Ravi Prabhakar More on email rm975@medschl.cam.ac.uk, ravipmore7@gmail.com

Contributors
Developed by: Dr. Ravi Prabhakar More

Conceptualized by: Dr. Ravi Prabhakar More, Dr. Varun Warrier, Dr. Chalres, Prof. Simon
